lib/DBSQLClient.ts

```diff
@@ -43,15 +45,19 @@ export default class DBSQLClient extends EventEmitter implements IDBSQLClient {
 
   private authProvider: IAuthentication;
 
+  private logger: IDBSQLLogger;
+
   private thrift = thrift;
 
-  constructor() {
+  constructor(logger?: IDBSQLLogger) {
```

Comment:
why do we want to expose `IDBSQLLogger` interface to the the end user?

instead we could only support passing a log level here. 

```suggestion
  constructor(logLevel?: enum) {
```

Reply:
I think it's not that bad idea to expose it 🙂 This will allow users to use log processing tools they like, e.g centry or others. I'd go even further and instead of creating default logger in `DBSQLClient`'s constructor just expose the default implementation as well - this will allow user to properly configure it (choose log level, destination - console or file, choose filename, etc.)

Reply:
Thanks @kravets-levko for the explanation. +1 
I started an internal conversation about this with all people here and @fjakobs

Reply:
So, the way it works presently is that the default logger is created only if the user hasn't submitted a logger of their own. Is there a better way?